core/new-gui/src/app/common/util/predicate.ts

```diff
@@ -0,0 +1,7 @@
+/**
+ * checks if the given parameter is undefined or not.
+ * @param x
+ * @returns {boolean}
+ */
+export function isDefined(x: any): boolean {return x !== undefined; }
```

Comment:
any reason we need such a trivial function here? also it's better to add a type guard

Reply:
This is just used as a predicate lambda. I don't want to write plain `xxx.filter((x)=>{x!==undefined})`. For the type guard, I think it can take in any value, do I need to change?